src/Table/read_ipac_table/read_ipac_header.cxx

```diff
@@ -28,105 +28,114 @@ void check_bar_position(std::vector<size_t> &bar_offsets, std::string &line,
 size_t tablator::Table::read_ipac_header(
         std::istream &ipac_file, std::array<std::vector<std::string>, 4> &columns,
         std::vector<size_t> &ipac_column_offsets) {
-    size_t current_line = 0;
+    size_t line_num = 0;
     char first_character = ipac_file.peek();
     for (; ipac_file && first_character == '\\'; first_character = ipac_file.peek()) {
         std::string line;
         std::getline(ipac_file, line);
-        ++current_line;
+        ++line_num;
         if (line.size() == 1) continue;
         if (line[1] == ' ') {
+            // Line starting with "\ " is a comment.
             comments.push_back(line.substr(2));
         } else {
+            // Line starting with "\X" (X not a space) should be keyword definitions of
+            // the form "\name = value".
             auto position_of_equal = line.find("=");
-            std::string key =
-                    boost::algorithm::trim_copy(line.substr(1, position_of_equal - 1));
-            std::string value =
-                    boost::algorithm::trim_copy(line.substr(position_of_equal + 1));
+            if (position_of_equal == std::string::npos) {
+                // Assume this is a comment missing the required space in position 1? JTODO
+                comments.push_back(line.substr(1));
+            } else {
+                std::string key = boost::algorithm::trim_copy(
+                        line.substr(1, position_of_equal - 1));
 
-            if (!boost::iequals(key, "fixlen") &&
-                !boost::iequals(key, "RowsRetrieved")) {
-                std::size_t first = value.find_first_not_of("\"'");
-                std::size_t last = value.find_last_not_of("\"'");
-                std::string value_substr;
-                if (first != std::string::npos)
-                    value_substr = value.substr(first, last - first + 1);
-
-                properties.emplace_back(key, Property(value_substr));
+                // These two properties/keywords are generated by
+                // write_ipac_table_header().
+                if (!boost::iequals(key, FIXLEN_KEYWORD) &&
+                    !boost::iequals(key, ROWS_RETRIEVED_KEYWORD)) {
+                    std::string value = boost::algorithm::trim_copy(
+                            line.substr(position_of_equal + 1));
+                    std::size_t first = value.find_first_not_of("\"'");
+                    if (first != std::string::npos) {
+                        std::size_t last = value.find_last_not_of("\"'");
+                        std::string value_substr =
+                                value.substr(first, last - first + 1);
+                        properties.emplace_back(key, Property(value_substr));
+                    }
```

Comment:
This isn't quite the same as what it is replacing - in particular, I think your version will not add empty valued keys, coming from lines like:
```
\foo= 
```
to `properties`. It's not obvious to me why you would change behavior for this case, so if this is intentional, I think a comment is warranted.

Reply:
Thanks, no, that wasn't intentional.  Will fix.